sysbuild/Kconfig.appcore

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+config SUPPORT_APPCORE
+	bool
+	default y if (SOC_NRF5340_CPUNET || SOC_NRF54H20_CPURAD)
+
+config APPCORE_REMOTE_BOARD_TARGET_CPUCLUSTER
+	string
+	default "cpuapp"
+
+config APPCORE_REMOTE_DOMAIN
+	string
+	default "CPUAPP" if APPCORE_REMOTE_BOARD_NAME != ""
+
+menu "Application core configuration"
+	depends on SUPPORT_APPCORE
+
+config SUPPORT_APPCORE_EMPTY
+	bool
+	default y
+
+config SUPPORT_APPCORE_REMOTE_SHELL
+	bool
+	default y
+
+config NRF_DEFAULT_EMPTY
+	bool
+
+config NRF_DEFAULT_REMOTE_SHELL
+	bool
+
+choice APPCORE
+	prompt "Appcore image"
+	default APPCORE_REMOTE_SHELL if NRF_DEFAULT_REMOTE_SHELL && APPCORE_REMOTE_BOARD_NAME != ""
+	default APPCORE_EMPTY if NRF_DEFAULT_EMPTY && APPCORE_REMOTE_BOARD_NAME != ""
+	depends on SUPPORT_APPCORE && !EXTERNAL_CONFIGURED_APPCORE
```

Comment:
I don't see the option `EXTERNAL_CONFIGURED_APPCORE` defined anywhere. Should it be added?

Reply:
No, that is a workaround for upstream samples that add a network core image in some builds, there aren't any that add an application core image

Reply:
So to clarify if a sample uses different appcore image (ie. DTM uses remote_hci) what is the proper way to configure it? Is it:
- Use the EXTERNAL_CONFIGURED_APPCORE
- or add the options and image names through Kconfig.sysbuild in sample directory?

Reply:
See second commit in PR